frontend/src/metabase/meta/types/Query.js

```diff
@@ -182,20 +182,13 @@ export type InsideFilter = [
   NumericLiteral,
   NumericLiteral,
 ];
-export type TimeIntervalFilter =
-  | [
-      "time-interval",
-      ConcreteField,
-      RelativeDatetimePeriod,
-      RelativeDatetimeUnit,
-    ]
-  | [
-      "time-interval",
-      ConcreteField,
-      RelativeDatetimePeriod,
-      RelativeDatetimeUnit,
-      FilterOptions,
-    ];
+export type TimeIntervalFilter = [
+  "time-interval",
+  ConcreteField,
+  RelativeDatetimePeriod,
+  RelativeDatetimeUnit,
+  ?TimeIntervalFilterOptions,
+];
```

Comment:
I'm not sure why the previous union type didn't work. Flow claimed that it couldn't pick a case, but the arity is different 🤷‍♂ 

Flow seems happier when I consolidated but made the options optional. Does that look ok?

Reply:
I actually don't think that's quite correct. It's saying you can have `null` in the last position, but leaving it out entirely will cause an error.

I'll see if I can figure out a solution.